### PR TITLE
better canvas gen

### PR DESCRIPTION
**Pull request by nathanielfernandes**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -27,8 +29,20 @@ var previewDir = mustGetEnvString("PREVIEW_DIR")
 var tempDir = getEnvWithFallback("TEMP_DIR", "./tmp")
 var previewBaseUrl = mustGetEnvString("PREVIEW_BASE_URL")
 
+// canvas api (quilt)
+var canvasBaseUrl = mustGetEnvString("CANVAS_BASE_URL")
+var canvasSecret = mustGetEnvString("CANVAS_SECRET")
+var previewGenCode = func() string {
```
Comment (robertkozin) [inline main.go]: Please use https://pkg.go.dev/embed to embed files into code

File: main.go [hunk 2]
```diff
@@ -44,6 +58,7 @@ func main() {
 	// Ensure preview dir exists
 	os.MkdirAll(previewDir, os.ModePerm)
 	os.MkdirAll(tempDir, os.ModePerm)
+	os.MkdirAll(tempDir, os.ModePerm)
```
Comment (robertkozin) [inline main.go]: Deplicate line

File: main.go [hunk 3]
```diff
@@ -60,6 +75,11 @@ func main() {
 	canvas.StartCanvasRunner()
 	spreview.StartPreviewRunner()
 
+	// start spotify runners
+	token.StartAccessTokenReferesher()
```
Comment (robertkozin) [inline main.go]: Duplicate lines

File: preview.ql [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+let x = w - @len(colors) * (s + spacing) - d
+let y = size + d
+
+for c in colors {
+    @DrawRectangle(x, y, s, s)
+    @SetColor(c)
+    @Fill()
+
+    x = x + s + spacing
+}
```
Comment (robertkozin) [inline preview.ql:44]: New line

**Review by robertkozin — CHANGES_REQUESTED** [review]
Can you also please look at the Dockerfile and increment the cache bust thing, you'll see it

**Comment by nathanielfernandes** [comment]
done 👍🏽

**Review by robertkozin — APPROVED** [review]

**Pull request merged**